src/evict/evict_lru.c

```diff
@@ -168,29 +171,27 @@ __wt_evict_server_wake(WT_SESSION_IMPL *session)
 }
 
 /*
- * __evict_thread_run --
- *	General wrapper for any eviction thread.
+ * __wt_evict_worker_run --
+ *	General wrapper for an eviction worker thread.
  */
-static WT_THREAD_RET
-__evict_thread_run(void *arg)
+int
+__wt_evict_worker_run(WT_SESSION_IMPL *session, WT_WORKER_THREAD *worker)
 {
 	WT_CACHE *cache;
 	WT_CONNECTION_IMPL *conn;
 	WT_DECL_RET;
-	WT_SESSION_IMPL *session;
 	bool did_work;
 
-	session = arg;
 	conn = S2C(session);
 	cache = conn->cache;
 
 #ifdef HAVE_DIAGNOSTIC
-	if (session == conn->evict_session)
-		WT_ERR(__wt_epoch(
-		    session, &cache->stuck_ts));	/* -Wuninitialized */
+	WT_ERR(__wt_epoch(session, &cache->stuck_ts));	/* -Wuninitialized */
```

Comment:
Should this only be initialized with `worker->id == 0`?  With this change, a reconfiguration to add a thread will unconditionally clear/reset the cache timestamp.